internal/repository/key.go

```diff
@@ -112,9 +112,26 @@ func OpenKey(ctx context.Context, s *Repository, name string, password string) (
 // given password. If none could be found, ErrNoKeyFound is returned. When
 // maxKeys is reached, ErrMaxKeysReached is returned. When setting maxKeys to
 // zero, all keys in the repo are checked.
-func SearchKey(ctx context.Context, s *Repository, password string, maxKeys int) (k *Key, err error) {
+func SearchKey(ctx context.Context, s *Repository, password string, maxKeys int, keyHint string) (k *Key, err error) {
 	checked := 0
 
+	if len(keyHint) > 0 {
```

Comment:
If you extract this into its own function, the error handling becomes nicer, since you can return an error and keep the happy path non-indented. Let me know if you'd like to see a sample.

Reply:
I disagree here, as error handling is not necessary, except to bail out of this block.  On any error, we want to proceed with business as usual and check every key.  The hint is only a hint.

Reply:
Fair enough.